### ISSUE TITLE
bug: between breaks if RDATE is an interval

### ISSUE THREAD

**Issue by natashamm** [niccokunzmann/python-recurring-ical-events#113]
**Describe the bug**
 

`between` breaks if RDATE is an interval. I'm not sure if it is an issues with `dateutil` or recurring events.

**To Reproduce**
 
```
import datetime
import recurring_ical_events
import pytz

today = datetime.datetime.now(pytz.timezone("America/Vancouver")).date()
future = today + datetime.timedelta(days=365)
for event in cal_i.walk("VEVENT") + cal_i.walk("VFREEBUSY"):
    if event.has_key('RDATE'):
        if (event.has_key('RDATE')):
            recurring_ical_events.of(event).between(today - datetime.timedelta(days=60), future)
```

**ICS file**
 

```
BEGIN:VCALENDAR
VERSION:2.0
X-WR-CALNAME;VALUE=TEXT:Test RDATE
BEGIN:VTIMEZONE
TZID:America/Vancouver
BEGIN:STANDARD
DTSTART:20221106T020000
TZOFFSETFROM:-0700
TZOFFSETTO:-0800
RDATE:20231105T020000
TZNAME:PST
END:STANDARD
BEGIN:DAYLIGHT
DTSTART:20230312T020000
TZOFFSETFROM:-0800
TZOFFSETTO:-0700
RDATE:20240310T020000
TZNAME:PDT
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
UID:1
DESCRIPTION:Test RDATE
DTSTART;TZID=America/Vancouver:20230920T120000
DTEND;TZID=America/Vancouver:20230920T140000
EXDATE;TZID=America/Vancouver:20231220T120000
RDATE;VALUE=PERIOD;TZID=America/Vancouver:20231213T120000/20231213T140000
RRULE:FREQ=MONTHLY;COUNT=9;INTERVAL=1;BYDAY=+3WE;B/YMONTH=1,2,3,4,5,9,10,11,
 12;WKST=MO
SUMMARY:Test RDATE
END:VEVENT

END:VCALENDAR
```

**Version:**
 
![2.0.2](https://raster.shields.io/badge/version-0.1.12b0-brightgreen.png)
 
- [ ] add an ICS file, example:
    https://github.com/niccokunzmann/python-recurring-ical-events/blob/f4a90b211f30bf0522f03514ba50eb69826f0fdb/test/calendars/issue-15-duplicated-events.ics#L1
- [ ] add a test, example:
    https://github.com/niccokunzmann/python-recurring-ical-events/blob/f4a90b211f30bf0522f03514ba50eb69826f0fdb/test/test_issue_15.py#L21
- [ ] fix the bug and ensure the test code passes

---
<small>
We're using [Polar.sh](https://polar.sh/niccokunzmann) so you can upvote and help fund this issue. We receive the funding once the issue is completed & confirmed by you. Thank you in advance for helping prioritize & fund our work.
</small>

<a href="https://polar.sh/niccokunzmann/python-recurring-ical-events/issues/113">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/113/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/113/pledge.svg">
</picture>
</a>

**Comment by niccokunzmann** [comment]
Hi @natashamm. I see that you have `EXDATE` in the calendar. This is not supported. If you think that is an issue, please open one!

**Comment by niccokunzmann** [comment]
@natashamm, this should be fixed in v2.1.0. Can you conform this to close this issue?

**Comment by natashamm** [comment]
Confirmed this is fixed. Thank you!
